src/haxelib/Util.hx

```diff
@@ -28,33 +28,41 @@ class Util {
 
 	macro static public function getHaxelibVersionLong() {
 		var version:String = readVersionFromHaxelibJson();
+		// check if the .git folder exist
+		// prevent getting the git info of a parent directory
+		if (!sys.FileSystem.isDirectory(".git"))
+			return macro $v{version};
+
 		var p;
 		try {
-			//check if the .git folder exist
-			//prevent getting the git info of a parent directory
-			if (!sys.FileSystem.isDirectory(".git"))
-				throw "Not a git repo.";
-
 			//get commit sha
 			p = new sys.io.Process("git", ["rev-parse", "HEAD"]);
-			final sha = p.stdout.readAll().toString().trim();
+			var sha = p.stdout.readAll().toString().trim();
 			p.close();
 
 			//check to see if there is changes, staged or not
 			p = new sys.io.Process("git", ["status", "--porcelain"]);
-			final changes = p.stdout.readAll().toString().trim();
+			var changes = p.stdout.readAll().toString().trim();
 			p.close();
 
-			version += switch(changes) {
+			var longVersion = version + switch(changes) {
 				case "":
 					' ($sha)';
 				case _:
 					' ($sha - dirty)';
 			}
-			return macro $v{version};
+			return macro $v{longVersion};
 		} catch(e:Dynamic) {
 			if (p != null) p.close();
 			return macro $v{version};
 		}
 	}
+
+	public static macro function replaceIfNull(toReplace:Expr, with:Expr)
```

Comment:
Let's please not do this. Duplicating expressions is not sound in the general case, so this is a potential trap.

Reply:
Reverted.